### PR TITLE
WIP: Glossary

### LINKED ISSUE #5275: DOC: Explain core MNE concepts in glossary.rst file
We have a few terms that are MNE specific, or for which multiple names would make sense, but we've settled on one. We don't have a doc page going through all of them, right?

Would it make sense to have such a page?

Thinking of:
- selection
- group_by
- picks
- raw
- epochs
- evoked
- info
- montage
- events (explain markers, triggers, stim channel ... here)
- ...

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
A glossary of MNE terms.

 closes #5275

### CODE REVIEW

**Comment by larsoner** [comment]
Seems reasonable to me. I would say add as many terms as you are willing to here (can be just what you have if you are sick of it already!) and we can expand incrementally.

One nice thing to add would be links to `sphx_glr_*` auto tutorials or examples.

**Comment by jona-sassenhagen** [comment]
>One nice thing to add would be links to sphx_glr_* auto tutorials or examples.

Yeah, though I'm not perfectly sure where to. Maybe the tutorials - or maybe the RawArray/EpochsArray ... APIs?

**Comment by larsoner** [comment]
Wherever makes sense for people to look for more information, if there are any such places.

**Comment by jona-sassenhagen** [comment]
I'm honestly not sure what a "projection" even is.

From where should I link to this?

**Comment by jona-sassenhagen** [comment]
https://8797-1301584-gh.circle-artifacts.com/0/html/glossary.html

Rendered

**Comment by agramfort** [comment]
I would put everything in alphabetical order. Wdyt?

**Comment by jona-sassenhagen** [comment]
Within categories? Maybe - right now it's structured "logically", but alphabetically would be more principled.

But still keep it in categories?

**Comment by agramfort** [comment]
For now until we have 50 items I think we can live without categories.

Each word can be a minimum section with a mini TOC at the top.

File: doc/glossary.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    Collection of metadata regarding a Raw, Epochs or Evoked object; e.g.,
+    channel locations, preprocessing history such as filters ...
+    See :ref:`sphx_glr_auto_tutorials_plot_info.py` for a narrative
+    overview.
+    
+events
+    A collection of Specific events occurring at specific time points; e.g.,
+    triggers, experimental condition events, ... MNE only knows about integer
+    events.
+
```
Comment (cbrnr) [inline doc/glossary.rst]: What about annotations? These are related to events, not sure if they are a subcategory of events or a separate entity.

**Comment by agramfort** [comment]
Annotations are a separate entity.

Annotations are defined by onset and duration in absolute time and can be
manually
edited / added by users eg to mark bad segments.

**Comment by jona-sassenhagen** [comment]
@agramfort like this?

**Comment by cbrnr** [comment]
I kind of liked the categories...

**Comment by jona-sassenhagen** [comment]
Me too :D Alex, can we vote on this? :)

**Comment by agramfort** [comment]
I will look at rendered version.

did you take inspiration from http://scikit-learn.org/dev/glossary.html ?
let's use the same structure / style.

**Comment by jona-sassenhagen** [comment]
Ok, that actually makes sense. I can just go by that.

**Comment by agramfort** [comment]
sure democracy sometimes work :)

but putting the content and then organising makes more sense to me.
Then it's just a matter of moving paragraphs

**Comment by cbrnr** [comment]
Standardization is good, but the categories in scikit-learn's glossary focus too much on the specific API, which MNE does not share: General concepts, Class APIs and Estimator Types, Target Types, Methods, Parameters, Attributes, Data and Sample Properties.

File: doc/glossary.rst [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+raw
+    Continuous data; dimensionality: sensors x times. May or may not be
+    preprocessed.
+    See :class:`RawArray <mne.io.RawArray>` for the API of the corresponding
+    object class, and :ref:`sphx_glr_auto_tutorials_plot_object_raw.py` for a
+    narrative overview.
+
+selection
+    A set of picks. E.g., all sensors included in a Region of Interest.
```
Comment (jasmainak) [inline doc/glossary.rst]: morphing?

Comment (jasmainak) [reply doc/glossary.rst]: source space?

File: doc/glossary.rst [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    channel locations, preprocessing history such as filters ...
+    See :ref:`sphx_glr_auto_tutorials_plot_info.py` for a narrative
+    overview.
+    
+montage
+    EEG channel names and the relative positions of the sensor w.r.t the scalp.
+    See :class:`mne.channels.Montage` for the API of the corresponding object
+    class.
+
+pick
```
Comment (jasmainak) [inline doc/glossary.rst]: it's just an index, no?

File: doc/glossary.rst [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    average over the trials of one subject for one condition, or over multiple
+    subjects.
+    See :class:`EvokedArray <mne.EvokedArray>` for the API of the corresponding
+    object class, and :ref:`sphx_glr_auto_tutorials_plot_object_evoked.py`
+    for a narrative overview.
+
+events
+    A collection of Specific events occurring at specific time points; e.g.,
+    triggers, experimental condition events, ... MNE only knows about integer
+    events. These can be stored as (3, n_events) numpy arrays, but are also
```
Comment (jasmainak) [inline doc/glossary.rst]: `(n_events, 3)`, no?

File: doc/glossary.rst [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+.. contents:: Contents
+   :local:
+
+
+MNE-Python core terminology
+===========================
+
+epochs
+    Epoched data; dimensionality: epochs x sensors x times. Typically,
```
Comment (jasmainak) [inline doc/glossary.rst]: `n_epochs x n_channels x n_times` to be consistent with `n_events`?

Comment (larsoner) [reply doc/glossary.rst]: I think everywhere we should stick to NumPy shape terminology, e.g.

Epoch data; shape (n_epochs, n_channels, n_time) ...

Comment (jasmainak) [reply doc/glossary.rst]: I agree ...

**Comment by jona-sassenhagen** [comment]
>Standardization is good, but the categories in scikit-learn's glossary focus too much on the specific API, which MNE does not share: General concepts, Class APIs and Estimator Types, Target Types, Methods, Parameters, Attributes, Data and Sample Properties.

Ok any more comments on @cbrnr 's suggestion here? @jasmainak ?

Would like to have the format clear before I redo the content :)

**Comment by agramfort** [comment]
Honestly I would first make sure every content concept is there. Any reformatting will take 10mins.

**Comment by jasmainak** [comment]
I agree with @agramfort 

The content should be there and accurate

**Comment by agramfort** [comment]
here is how the glossary looks like:

https://9032-1301584-gh.circle-artifacts.com/0/html/glossary.html

do you see anything to add / clarify?

**Comment by jasmainak** [comment]
Looks nice! Maybe add `trans`? I remember being confused by it early on, since it was an abbreviation. 

Also maybe `first_samp`? Not sure if glossary is the right place for it but I'm still getting user reports that they don't understand why it's not documented ...

**Comment by jasmainak** [comment]
Travis is failing ...

Otherwise LGTM

**Comment by larsoner** [comment]
Fixed the spelling errors that were causing the failure, and added `trans`. Let's get this in and iterate further in `master` as we find more items to add.

Thanks @jona-sassenhagen !

**Comment by jasmainak** [comment]
Nice addition!

**Pull request merged**